### PR TITLE
Do not cache the settings/sections in the database anymore

### LINKED ISSUE #6382: {admin,personal}_sections and {admin,personal}_settings are cached inside DB
The `appinfo/info.xml` file can be [used to extend](https://github.com/nextcloud/server/blob/2cb346114685ded69f5291812503c278298c1bce/apps/files_external/appinfo/info.xml#L29-L34) the `Personal Settings` and `Admin Settings` page:

```xml
<settings>
	<admin>OCA\Files_External\Settings\Admin</admin>
	<admin-section>OCA\Files_External\Settings\Section</admin-section>
	<personal>OCA\Files_External\Settings\Personal</personal>
	<personal-section>OCA\Files_External\Settings\PersonalSection</personal-section>
</settings>
```

What's the reason these options are also stored (cached?) inside the database?
This has already caused quite some confusion as these entries apparently only get refreshed if the app is updated. For example adjusting the `getPriority()` / `getSection()` return value of a `ISettings`-implementing class has no effect unless the app is updated.

Another problem is that old entries are not removed. Steps to reproduce: Remove some of the elements in `<settings>` and upgrade the app: The rows are still in the DB.

### PR DESCRIPTION
**Pull request by nickvergessen**
This caused more troubles then it had benefits, especially
when an app got disabled or was removed without being disabled.

Fix #6382

### CODE REVIEW

File: lib/public/Settings/IManager.php [hunk 1]
```diff
@@ -49,62 +49,34 @@
...
-	 *                        by the KEY_ constant of  this interface. The value
-	 *                        must always be a class name, implement either
-	 *                        IAdmin or ISection. I.e. only one section and admin
-	 *                        setting can be configured per app.
-	 * @since 9.1.0
+	 * @param string $type 'admin' or 'personal'
+	 * @param string $section Class must implement OCP\Settings\ISection
+	 * @since 14.0.0
 	 */
-	public function setupSettings(array $settings);
```
Comment (nickvergessen) [inline lib/public/Settings/IManager.php]: In theory this breaks the public API, but apps should only register via info.xml anyway and that still works as before

Comment (MorrisJobke) [reply lib/public/Settings/IManager.php]: None of the appstore apps uses this.

Comment (rullzer) [reply lib/public/Settings/IManager.php]: Kill it !

**Review by MorrisJobke — APPROVED** [review]
Tested, works and the code looks good 👍

**Comment by MorrisJobke** [comment]
Let me rebase it due to the conflict

**Review by rullzer — APPROVED** [review]

**Pull request merged**